### PR TITLE
lfx pre test adding expression syntax rules and refine test cases

### PR DESCRIPTION
**Pull request by Vishalk91-4**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

re #8 

#### 2. What is the scope of this PR (e.g. component or file name):

In the `test/corpus` I did add test cases for unary expressions, conditonal expressions and slice expressions.

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [ ] Affects user behaviors
- [x] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [ ] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

### CODE REVIEW

File: grammar.js [hunk 1]
```diff
@@ -107,6 +107,7 @@ const CONSTANTS = {
   FALSE: "False",
   NONE: "None",
   UNDEFINED: "Undefined",
+  NULL: "Null",
```
Comment (Peefy) [inline grammar.js:110]: KCL has no `Null` keyword.

File: grammar.js [hunk 2]
```diff
@@ -127,6 +128,11 @@ const PREC = {
   as: 23,
 };
 
+const STRING_DELIMITERS = {
+  QUOTED: '"',
+  RAW: '`',
```
Comment (Peefy) [inline grammar.js:133]: KCL has no ` token.

**Comment by Peefy** [comment]
You can refer the following docs, impls and avoid submitting syntax unrelated to KCL.

+ KCL Tour https://www.kcl-lang.io/docs/reference/lang/tour
+ KCL Spec https://www.kcl-lang.io/docs/reference/lang/spec/
+ KCL Core Parser: https://github.com/kcl-lang/kcl/tree/main/kclvm/parser/src

File: grammar.js [hunk 3]
```diff
@@ -474,6 +491,24 @@ module.exports = grammar({
         field('right', $.primary_expression),
       ))));
     },
+    
+    advanced_operator: $ => {
```
Comment (Peefy) [inline grammar.js:495]: KCL has no these operators.

File: grammar.js [hunk 4]
```diff
@@ -517,6 +552,29 @@ module.exports = grammar({
       field('right', $.expression),
     ),
 
+    function_call_assignment: $ => seq(
```
Comment (Peefy) [inline grammar.js:555]: KCL has no call assignment stmt.

File: grammar.js [hunk 5]
```diff
@@ -657,10 +715,17 @@ module.exports = grammar({
       $.expression,
     )),
 
-    string: $ => seq(
-      $.string_start,
-      repeat($.string_content),
-      $.string_end,
+    ternary_expression: $ => prec.right(PREC.conditional, seq(
```
Comment (Peefy) [inline grammar.js:718]: KCL has no ternary_expression.

**Comment by Vishalk91-4** [comment]
> You can refer the following docs, impls and avoid submitting syntax unrelated to KCL.
> 
> * KCL Tour https://www.kcl-lang.io/docs/reference/lang/tour
> * KCL Spec https://www.kcl-lang.io/docs/reference/lang/spec/
> * KCL Core Parser: https://github.com/kcl-lang/kcl/tree/main/kclvm/parser/src

Sorry, mistakenly overlooked that
Will change things nd submit those relatd to KCL

**Comment by Vishalk91-4** [comment]
> You can refer the following docs, impls and avoid submitting syntax unrelated to KCL.
> 
> * KCL Tour https://www.kcl-lang.io/docs/reference/lang/tour
> * KCL Spec https://www.kcl-lang.io/docs/reference/lang/spec/
> * KCL Core Parser: https://github.com/kcl-lang/kcl/tree/main/kclvm/parser/src

Just to confirm
Do we have to add the remaining expression, statements,.. in `grammar.js` and add the tests for them in `test/corpus/`

**Comment by Peefy** [comment]
> Just to confirm @Peefy Do we have to add the remaining expression, statements,.. in `grammar.js` and add the tests for them in `test/corpus/`

Yes. Different expressions and syntax are added to `grammar.js`, which are relatively independent. Add/modify syntax one by one and add test cases in `test/corpus/`. You can view the pass status and diff of test cases in CI https://github.com/kcl-lang/tree-sitter-kcl/actions/runs/9085992382/job/24979088541?pr=10

**Comment by Daksh-10** [comment]
Is there any documentation of what should be the output for the expressions when the syntax tree is drawn for them. For example,
image
Here, how would we know, that we need to specify a string like this for 'kcl' language.

**Comment by Peefy** [comment]
Hello @Daksh-10 , do you mean the tree-sitter grammar for the KCL string? https://github.com/kcl-lang/tree-sitter-kcl/blob/main/grammar.js#L660

If you want to know how tree-sitter grammar syntax and principle, you can refer https://tree-sitter.github.io/

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**